### PR TITLE
adds quoting style option to csv writer config

### LINKED ISSUE #1254: configure quoting for csv writers
Allow for full and minimal quoting to be configurable for both (arrow and Python) csv writers.
1. Add new config parameter
2. Keep the current as default
3. Confgure writers with it
4. Update tests and documentation

### PR DESCRIPTION
**Pull request by rudolfix**
### Description
Closes #1254

### CODE REVIEW

File: dlt/common/data_writers/writers.py [hunk 1]
```diff
@@ -398,23 +402,32 @@ def __init__(
...
         self.include_header = include_header
         self.delimiter = delimiter
+        self.quoting: CsvQuoting = quoting
         self.writer: csv.DictWriter[str] = None
         self.bytes_encoding = bytes_encoding
 
     def write_header(self, columns_schema: TTableSchemaColumns) -> None:
         self._columns_schema = columns_schema
+        if self.quoting == "quote_needed":
+            quoting: Literal[1, 2] = csv.QUOTE_NONNUMERIC
```
Comment (sultaniman) [inline dlt/common/data_writers/writers.py:418]: `csv.QUOTE_NONNUMERIC` has the value of `2` and `1` corresponds to `csv.QUOTE_ALL` should we change this to int from `Literal`?

File: dlt/destinations/impl/postgres/postgres.py [hunk 2]
```diff
@@ -112,6 +112,10 @@ def __init__(self, table_name: str, file_path: str, sql_client: Psycopg2SqlClien
         with FileStorage.open_zipsafe_ro(file_path, "rb") as f:
             # all headers in first line
             headers = f.readline().decode("utf-8").strip()
+            # quote headers if not quoted - all special keyword like "binary" must be quoted
+            headers = ",".join(
+                h if h.startswith('"') else '"' + h + '"' for h in headers.split(",")
```
Comment (sultaniman) [inline dlt/destinations/impl/postgres/postgres.py:117]: I think we need to use `f`-strings for readbility

**Review by sh-rp — DISMISSED** [review]

File: dlt/common/data_writers/writers.py (hunk 1 above)
Comment (sultaniman) [reply dlt/common/data_writers/writers.py:418]: so I did a read up I thought, type information would be erased and become any in the case of `else` branch but according to mypy it reads the first type definition for a variable and later uses it to validate the assigned type.

File: dlt/destinations/impl/postgres/postgres.py (hunk 2 above)
Comment (rudolfix) [reply dlt/destinations/impl/postgres/postgres.py:117]: OK I agree, will change it

**Pull request merged**